tools/src/main/java/com/nedap/archie/flattener/Flattener.java

```diff
@@ -146,7 +147,11 @@ public Archetype flatten(Archetype toFlatten) {
 
         if(parent.getParentArchetypeId() != null) {
             //parent needs flattening first
-            parent = getNewFlattenerForParent().flatten(parent);
+            Flattener parentFlattener = getNewFlattenerForParent();
+            parent = parentFlattener.flatten(parent);
+            parentFlattener.getRepository().getExtraArchetypes().forEach(
```

Comment:
Sounds sensible to do this. Some comments on what this line does is very welcome.